esti/gc_test.go

```diff
@@ -169,17 +265,49 @@ func prepareForGC(t *testing.T, ctx context.Context, testCase testCase, blocksto
 	return commitId
 }
 
-func getGCMode(testCase testCase) int {
-	if testCase.testMode == "mark" {
-		return markOnlyMode
-	} else if testCase.testMode == "sweep" {
-		return sweepOnlyMode
-	} else {
-		return fullGCMode
+func getSparkSubmitArgs() []string {
+	return []string{
+		"--master", "spark://localhost:7077",
+		"--conf", "spark.driver.extraJavaOptions=-Divy.cache.dir=/tmp -Divy.home=/tmp",
+		"--conf", "spark.hadoop.lakefs.api.url=http://lakefs:8000/api/v1",
+		"--conf", "spark.hadoop.fs.s3.impl=org.apache.hadoop.fs.s3a.S3AFileSystem",
+		"--conf", "spark.hadoop.lakefs.api.access_key=AKIAIOSFDNN7EXAMPLEQ",
+		"--conf", "spark.hadoop.lakefs.api.secret_key=wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY",
+		"--class", "io.treeverse.clients.GarbageCollector",
+		"--packages", "org.apache.hadoop:hadoop-aws:3.2.4",
+	}
+}
+
+func getDockerArgs(workingDirectory string) []string {
+	return []string{"run", "--network", "host", "--add-host", "lakefs:127.0.0.1",
+		"-v", fmt.Sprintf("%s/ivy:/opt/bitnami/spark/.ivy2", workingDirectory),
+		"-v", fmt.Sprintf("%s:/opt/metaclient/client.jar", metaclientJarPath),
+		"--rm",
+		"-e", "AWS_ACCESS_KEY_ID",
+		"-e", "AWS_SECRET_ACCESS_KEY",
+	}
+}
+
+func runGC(t *testing.T, repo string, extraSparkArgs ...string) {
+	workingDirectory, err := os.Getwd()
+	if err != nil {
+		t.Fatal("Getting working directory: ", err)
+	}
+	workingDirectory = strings.TrimSuffix(workingDirectory, "/")
+	dockerArgs := getDockerArgs(workingDirectory)
+	dockerArgs = append(dockerArgs, "docker.io/bitnami/spark:3.2", "spark-submit")
```

Comment:
shouldn't the docker image be injected?

Reply:
Opened issue for it: https://github.com/treeverse/lakeFS/issues/4998